app/actions/v3/service_instance_update_managed.rb

```diff
@@ -338,6 +339,15 @@ def raise_if_cannot_update!
         end
       end
 
+      def raise_if_invalid_state!
+        raise CloudController::Errors::ApiError.new_from_details('ServiceInstanceNotFound', service_instance.name) if service_instance.create_failed?
+        raise InvalidServiceInstance.new('The service instance is getting deleted or its deletion failed.') if service_instance.delete_in_progress? ||
```

Comment:
Instead of raising an `InvalidServiceInstance` error that needs to be caught in the `ServiceInstancesController`, you could raise an `UnprocessableUpdate` error similar to the other `raise_if_...` methods:
```
raise UnprocessableUpdate.new_from_details('ServiceInstanceInvalid', 'The service instance is getting deleted or its deletion failed.')
```

I like the `do_something if some_condition` syntax, but when the line gets too long, you should switch to the 'normal' `if ... end` syntax.

Reply:
I changed it accordingly.